RealmSwift/SwiftUI.swift

```diff
@@ -0,0 +1,395 @@
+ Bound types are observed, and write transactions are implicit.
+ */
+
+// MARK: Bound Types
+
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *)
+protocol Bound: ObservableObject where ObjectWillChangePublisher == ObservableObjectPublisher {
+    associatedtype T
+
+    var value: T { get }
+
+    init(_ value: T)
+
+    func observe(binder: inout RealmBind<T>, _ block: @escaping (RealmBind<T>) -> Void) -> NotificationToken
+}
+
+@dynamicMemberLookup
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *)
+final class BoundObject<T: Object> : Bound {
+    public var value: T
+
+    public required init(_ value: T) {
+        self.value = value
+    }
+
+    public subscript<V>(dynamicMember member: ReferenceWritableKeyPath<T, V>) -> Binding<V> {
+        get {
+            Binding(get: {
+                self.value[keyPath: member]
+            }, set: { newValue in
+                try! self.value.realm!.write {
+                    self.value[keyPath: member] = newValue
+                }
+            })
+        }
+    }
+
+    // if property is a list, we want to freeze it before fetching it
+    public subscript<V>(dynamicMember member: ReferenceWritableKeyPath<T, V>) -> Binding<V> where V: ListBase {
+        get {
+            Binding(get: {
+                self.value[keyPath: member].freeze()
+            }, set: {
+                self.value[keyPath: member] = $0
+            })
+        }
+    }
+
+    func observe(binder: inout RealmBind<T>, _ block: @escaping (RealmBind<T>) -> Void) -> NotificationToken {
+        return value.observe { [binder] _ in
+            block(binder)
+        }
+    }
+}
+
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *)
+final class BoundList<Element: RealmCollectionValue>: ObservableObject {
+    public typealias T = Binding<List<Element>>
+    public var value: Binding<List<Element>>
+    public required init(_ value: Binding<List<Element>>) {
+        self.value = value
+    }
+
+    public subscript<V>(dynamicMember member: ReferenceWritableKeyPath<T, V>) -> Binding<V> {
+        get {
+            Binding(get: {
+                self.value[keyPath: member]
+            }, set: { newValue in
+                try! self.value.wrappedValue.realm!.write {
+                    self.value[keyPath: member] = newValue
+                }
+            })
+        }
+    }
+
+    // if property is a list, we want to freeze it before fetching it
+    public subscript<V>(dynamicMember member: ReferenceWritableKeyPath<T, V>) -> Binding<V> where V: ListBase {
+        get {
+            Binding(get: {
+                self.value[keyPath: member].freeze()
+            }, set: {
+                self.value[keyPath: member] = $0
+            })
+        }
+    }
+}
+
+@available(OSX 10.15, watchOS 6.0, iOS 13.0, iOSApplicationExtension 13.0, OSXApplicationExtension 10.15, tvOS 13.0, *)
+final class BoundResults<Element: RealmCollectionValue>: ResultsBase, Bound {
```

Comment:
It seems like BoundResults could be replaced entirely with an extension on RLMResults.